### PR TITLE
Adds Low-level Rust Bindings for Ion C

### PR DESCRIPTION
**Pull request by almann**
Addresses #34.

* Makes the top-level `Cargo.toml` a workspace.
* Adds the `ion-c-sys` sub-crate to the workspace.
  - Adds `ion-c` submodule.
  - Adds a `build.rs` to build/install/link statically to `ion-c`.
  - Adds `bindgen` generated bindings.
  - Adds rudimentary tests for basic `ion-c` functionality.
* Adds documentation in `README.md` to describe how to build/test (using `--workspace`).
* Updates Github Action workflow to support `bindgen` and workspace build/tests.

This change **does not**:

* Add easier to use Rust APIs nor implements Ion Rust's cursor API.
* Have comprehensive testing for the Rust bindings themselves.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ion-c-sys/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    fn read_ion_null() {
+        let input = CString::new("null").unwrap();
+        let buf = input.as_ptr() as *mut u8;
+        let buf_size = input.as_bytes_with_nul().len() as i32;
+        let mut ion_reader = ptr::null_mut();
+        let mut ion_type = ptr::null_mut();
+        let mut ion_type2 = ptr::null_mut();
+        let mut mybool: BOOL = 0;
+
+        // let mut ion_value = 0;
```
Comment (therapon) [inline ion-c-sys/src/lib.rs:30]: nit: leftover?

File: ion-c-sys/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    fn read_ion_timestamp_null() {
+        let input = CString::new("null.timestamp").unwrap();
+        let buf = input.as_ptr() as *mut u8;
+        let buf_size = input.as_bytes_with_nul().len() as i32;
+        let mut ion_reader = ptr::null_mut();
+        let mut ion_type = ptr::null_mut();
+        let mut ion_type2 = ptr::null_mut();
+        let mut mybool: BOOL = 0;
+
+        // let mut ion_value = 0;
```
Comment (therapon) [inline ion-c-sys/src/lib.rs:53]: nit: ditto?

File: ion-c-sys/src/lib.rs (hunk 1 above)
Comment (almann) [reply ion-c-sys/src/lib.rs:30]: Yep--let me kill that.

File: ion-c-sys/src/lib.rs (hunk 2 above)
Comment (almann) [reply ion-c-sys/src/lib.rs:53]: Yep--let me kill that.

**Pull request merged**